### PR TITLE
Sync latest changes from the ReClaim template

### PR DESCRIPTION
**Pull request by yurii-ve**
- lib folder
- `getStaticRoutes`
- user session

### CODE REVIEW

File: app/routes/_index/route.tsx [hunk 1]
```diff
@@ -1,19 +1,19 @@
...
 
 export const loader = async ({ request }: LoaderFunctionArgs) => {
-    const productsResponse = await getEcomApi().getPromotedProducts();
+    const ecomApi = await initializeEcomApi(request);
+    const productsResponse = await ecomApi.getPromotedProducts();
     if (productsResponse.status === 'failure') {
         throw json(productsResponse.error);
     }
 
-    return { products: productsResponse.body, canonicalUrl: getUrlOriginWithPath(request.url) };
```
Comment (olehrakwix) [inline app/routes/_index/route.tsx:16]: shouldn't we have this canonical url now?

Comment (yurii-ve) [reply app/routes/_index/route.tsx:16]: we should.. adding

File: app/routes/_index/route.tsx [hunk 2]
```diff
@@ -63,11 +63,6 @@ export const meta: MetaFunction<typeof loader> = ({ data }) => {
             name: 'description',
             content: description,
         },
-        {
-            tagName: 'link',
```
Comment (olehrakwix) [inline app/routes/_index/route.tsx]: is this expected?

File: app/routes/about/route.tsx [hunk 3]
```diff
@@ -38,11 +32,6 @@ export const meta: MetaFunction<typeof loader> = ({ data }) => {
             name: 'description',
             content: description,
         },
-        {
-            tagName: 'link',
```
Comment (olehrakwix) [inline app/routes/about/route.tsx]: same here

Comment (yurii-ve) [reply app/routes/about/route.tsx]: added everywhere.. nice catch 👍

File: src/components/product-filters/product-filters.tsx [hunk 4]
```diff
@@ -1,8 +1,8 @@
 import { useCallback, useMemo } from 'react';
-import { productFiltersFromSearchParams, searchParamsFromProductFilters } from '~/api/product-filters';
-import { IProductFilters } from '~/api/types';
-import { formatPrice, mergeUrlSearchParams } from '~/utils';
-import { useSearchParamsOptimistic } from '~/utils/use-search-params-optimistic';
+import { productFiltersFromSearchParams, searchParamsFromProductFilters } from '~/lib/ecom';
+import { IProductFilters } from '~/lib/ecom';
```
Comment (tetianaman) [inline src/components/product-filters/product-filters.tsx]: FYI: small warning
Combine imports from '~/lib/ecom';

**Review by tetianaman — APPROVED** [review]

**Review by alisey — APPROVED** [review]
Looks fine. It's going to be fine.

File: app/routes/category.$categorySlug/route.tsx [hunk 5]
```diff
@@ -21,18 +26,18 @@ export const loader = async ({ params, request }: LoaderFunctionArgs) => {
         throw new Error('Missing category slug');
     }
 
-    const api = getEcomApi();
+    const ecomApi = await initializeEcomApi(request);
```
Comment (alisey) [inline app/routes/category.$categorySlug/route.tsx:29]: I think `getProductsRouteData()` implements this entire loader, but it's ok.

**Comment by alisey** [comment]
@PeterShershov FYI

**Review by olehrakwix — APPROVED** [review]

**Pull request merged**